canonical/ubuntu-pro-client
# tests: behave-any anywhere

## Why is this needed?

> And with everyone ~~super~~`behave-any`, no one will be!
>     - Syndrome (The Incredibles)

This extends the "behave-any" refactor to all integration tests. That means any combination of tests, regardless of the cloud/platform they run on, can be run with one command. After all tests have this support, it opens up a few simplifications, including removing all `@uses.config.machine_type.*` tags from all tests since they're no longer needed for filtering. This also allows consolidation of some tests since tests can now be parametrized by machine_type. I didn't perform all consolidation possible, only a few for now as demonstration.

In this PR, I'm maintaining the capability to run the tests in a per release, per platform matrix via the run-integration-tests.py script. However, after this refactor we don't have to split the tests up that way. One thing we could do instead is split the tests up by feature, and run whole feature files together.

Jenkins PR to go along with this one is here: https://github.com/canonical/server-jenkins-jobs/pull/292

## Test Steps
- CI should still pass and should run the same sets of tests that it did before
- try running a test across multiple platforms: `tox -e behave -- -n "Proxy auto-attach on a cloud Ubuntu Pro machine"`

## Checklist


## Does this PR require extra reviews?
